### ISSUE TITLE
response_model works wrong when using default_factory

### ISSUE THREAD

**Issue by blurhead** [tiangolo/fastapi#5486]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List

from fastapi import FastAPI
from pydantic import BaseModel, Field

app = FastAPI()


class Schema(BaseModel):
    business: List[str] = Field(default_factory=list)


@app.get("/test", response_model=Schema)
def test():
    return {}
```

### Description

expect: {"business": []}

actually: {"business": None}

### Operating System

Linux

### FastAPI Version

0.81.0

### Python Version

3.8

**Comment by axel584** [comment]
@blurhead  : If you want the default value to be an empty list, why don't you specify it like this:

```
class Schema(BaseModel):
    business: List[str] = []
```

Using the default_factory allows you to "generate" a value that will be different for each instance (like a uuid or a date)

**Comment by blurhead** [comment]
> @blurhead : If you want the default value to be an empty list, why don't you specify it like this:
> 
> ```
> class Schema(BaseModel):
>     business: List[str] = []
> ```
> 
> Using the default_factory allows you to "generate" a value that will be different for each instance (like a uuid or a date)

set default value to [] is dangerous, cause it's mutable, changed once, effect everywhere

**Comment by blurhead** [comment]
duplicate with https://github.com/tiangolo/fastapi/issues/3198, which was created at 2020, so no plan to fix this?

**Comment by Dustyposa** [comment]
You can try this one:
```python
class Schema(BaseModel):
    business: List[str] = Field([])
```

**Comment by JarroVGIT** [comment]
@blurhead 
> set default value to [] is dangerous, cause it's mutable, changed once, effect everywhere

That is true, when you provide a default value in the constructor of a class. But when dealing with pydantic models, it is perfectly fine to use a mutable as a default value for a property. See the below example to illustrate:

```python
from pydantic import BaseModel, Field

class Example_1(BaseModel):
    id: str
    l: list[int | str] = []

a = Example_1(id="a")
b = Example_1(id="b")

a.l.append("from_a")
print("From example 1 based on BaseModel:")
print(a)
print(b)
# -----------------------------------------------------
class Example_2:
    def __init__(self, id: str, l: list[str | int] = []) -> None:
        self.id = id
        self.l = l

a = Example_2(id="a")
b = Example_2(id="b")

a.l.append("from_a")
print("From example 2 based on constructor:")
print(a.l)
print(b.l)
```

This will result in:
```
From example 1 based on BaseModel:
id='a' l=['from_a']
id='b' l=[]
From example 2 based on constructor:
['from_a']
['from_a']
```

**Comment by blurhead** [comment]
> @blurhead 
> 
> > set default value to [] is dangerous, cause it's mutable, changed once, effect everywhere
> 
> 
> 
> That is true, when you provide a default value in the constructor of a class. But when dealing with pydantic models, it is perfectly fine to use a mutable as a default value for a property. See the below example to illustrate:
> 
> 
> 
> ```python
> 
> from pydantic import BaseModel, Field
> 
> 
> 
> class Example_1(BaseModel):
> 
>     id: str
> 
>     l: list[int | str] = []
> 
> 
> 
> a = Example_1(id="a")
> 
> b = Example_1(id="b")
> 
> 
> 
> a.l.append("from_a")
> 
> print("From example 1 based on BaseModel:")
> 
> print(a)
> 
> print(b)
> 
> # -----------------------------------------------------
> 
> class Example_2:
> 
>     def __init__(self, id: str, l: list[str | int] = []) -> None:
> 
>         self.id = id
> 
>         self.l = l
> 
> 
> 
> a = Example_2(id="a")
> 
> b = Example_2(id="b")
> 
> 
> 
> a.l.append("from_a")
> 
> print("From example 2 based on constructor:")
> 
> print(a.l)
> 
> print(b.l)
> 
> ```
> 
> 
> 
> This will result in:
> 
> ```
> 
> From example 1 based on BaseModel:
> 
> id='a' l=['from_a']
> 
> id='b' l=[]
> 
> From example 2 based on constructor:
> 
> ['from_a']
> 
> ['from_a']
> 
> ```

I tried this example, seems that pydantic will automatically copy mutable object, using [] solves my problem, thx
